File: fetch_ig.js
const https = require('https');
const fs = require('fs');

const items = [
  { url: 'https://www.instagram.com/reel/C-p913RSF8U/', name: 'sccg-1' },
  { url: 'https://www.instagram.com/reel/C9t-zVoyQ-S/', name: 'sccg-2' },
  { url: 'https://www.instagram.com/reel/C_iHRHAttW6/', name: 'sccg-3' },
  { url: 'https://www.instagram.com/reel/DKrgdzXN4uY/', name: 'kifs-1' },
  { url: 'https://www.instagram.com/reel/DI3-m4wTKC_/', name: 'kifs-2' }
];

async function fetchMeta(url) {
  return new Promise((resolve, reject) => {
    https.get(url, { headers: { 'User-Agent': 'curl/7.68.0' } }, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => {
        // Find og:image
        const match = data.match(/<meta property="og:image" content="([^"]+)"/);
        if (match) {
          // Unescape HTML entities
          const imgUrl = match[1].replace(/&amp;/g, '&');
          resolve(imgUrl);
        } else {
          resolve(null);
        }
      });
    }).on('error', reject);
  });
}

async function downloadImage(url, dest) {
  return new Promise((resolve, reject) => {
    const file = fs.createWriteStream(dest);
    https.get(url, (res) => {
      res.pipe(file);
      file.on('finish', () => {
        file.close(resolve);
      });
    }).on('error', err => {
      fs.unlink(dest, () => {});
      reject(err);
    });
  });
}

async function run() {
  for (let i = 0; i < items.length; i++) {
    const item = items[i];
    console.log(`Fetching ${item.url}...`);
    try {
      const imgUrl = await fetchMeta(item.url);
      if (imgUrl) {
        console.log(`Found image, downloading to ${item.name}.jpg`);
        await downloadImage(imgUrl, `./img/gallery/${item.name}.jpg`);
        console.log(`Success: ${item.name}`);
      } else {
        console.log(`No image found for ${item.url}`);
      }
    } catch (e) {
      console.error(`Failed ${item.url}: ${e.message}`);
    }
  }
}

run();

File: fetch_ig.py
import urllib.request
import re
import sys

def get_ig_image(url, output_path):
    try:
        req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'})
        html = urllib.request.urlopen(req).read().decode('utf-8')
        match = re.search(r'<meta property="og:image" content="([^"]+)"', html)
        if match:
            img_url = match.group(1).replace('&amp;', '&')
            urllib.request.urlretrieve(img_url, output_path)
            print(f"Success: {output_path}")
        else:
            print(f"Failed to find og:image for {url}")
    except Exception as e:
        print(f"Error for {url}: {e}")

urls = [
    ("https://www.instagram.com/reel/DTcR_lgjH4d/", "img/gallery/vivek-triumph.png"),
    ("https://www.instagram.com/p/DCGQE1myElC/", "img/gallery/shanvi-abudhabi.png"),
    ("https://www.instagram.com/reel/C8mTBM8ydA_/", "img/gallery/shivali-lbp.png"),
    ("https://www.instagram.com/reel/DEJ993Mqlsk/", "img/gallery/reet-miniso.png"),
    ("https://www.instagram.com/reel/C_h_ReAJBSM/", "img/gallery/reet-yousta.png"),
    ("https://www.instagram.com/reel/DBL2qZPyFZv/", "img/gallery/shruti-oppo.png"),
    ("https://www.instagram.com/reel/CxFQx5mRgHf/", "img/gallery/shreya-ytmusic.png"),
    ("https://www.instagram.com/reel/DBYUPYjI2qh/", "img/gallery/shruti-sugar.png"),
    ("https://www.instagram.com/reel/DBx7Y3Lo17z/", "img/gallery/shruti-kitkat.png"),
    ("https://www.instagram.com/reel/DC9JAZKSCKx/", "img/gallery/shruti-uber.png")
]

for url, path in urls:
    get_ig_image(url, path)
